media/js/mozorg/contribute/taskview.js

```diff
@@ -0,0 +1,312 @@
+
+    /**
+     * Handles blur and focus events on the main window, and completes
+     * the task step once the main window receives focus.
+     */
+    function handleFocusChange($step) {
+        $(window).one('focus', function() {
+            completeStep($step);
+        });
+    }
+
+    /**
+     * Sends data to GA about the interaction steps a user has taken.
+     * @param {string} interaction - The kind of interaction
+     * @param {int} variation - Whether it is a v1 or v2 task.
+     */
+    function trackInteraction(interaction, variation) {
+        window.dataLayer.push({
+            event: 'get-involved-task-interaction',
+            interaction: interaction,
+            variation: variation
+        });
+    }
+
+    /**
+     * Initializes the tweet form, set's up the character counter and binds
+     * the submit event handler.
+     */
+    function initTweetForm() {
+        var $tweetField = $('#tweet_txt');
+        var $charCount = $('.char-count');
+        var maxLength = 140;
+        var tweetLength = $tweetField.val().length;
+
+        // get and show the initial number of characters
+        $charCount.text(maxLength - tweetLength);
+
+        $tweetField.on('keyup', function() {
+            tweetLength = $tweetField.val().length;
+            $charCount.text(maxLength - tweetLength);
+        });
+
+        $signupTweetForm.on('submit', function(event) {
+            event.preventDefault();
+
+            var $submitButton = $('input[type="submit"]');
+            var tweetIntentURL = $signupTweetForm.attr('action');
+            var tweetContent = $tweetField.val();
+            var hashTag = $('#hashtag').attr('value');
+            var tweet = encodeURI(tweetIntentURL + '?text=' + tweetContent + '&hashtags=' + hashTag);
+
+            window.open(tweet, 'twitter', 'width=550,height=480,scrollbars');
+
+            handleFocusChange($submitButton);
+        });
+    }
+
+    /**
+     * Handles completion of Whimsy interaction steps.
+     */
+    function whimsy(event) {
+
+        var $this = $(event.target);
+
+        if ($this.data('action') === 'install') {
+            handleVisibilityChange($this);
+            trackInteraction('install whimsy', $this.data('variation'));
+        } else if ($this.data('action') === 'rate') {
+            handleVisibilityChange($this);
+            trackInteraction('AMO exit link', $this.data('variation'));
+        }
+    }
+
+    /**
+     * Handles completion of Firefox Mobile interaction steps.
+     */
+    function installFirefox(event) {
+        var $this = $(event.target);
+
+        if ($this.data('action') === 'install') {
+            handleVisibilityChange($this);
+        }
+    }
+
+    /**
+     * Handles completion of Follow Mozilla interaction steps.
+     */
+    function followMozilla(event) {
+        var $this = $(event.target);
+        var intentURL = $this.attr('href');
+        var screenName = intentURL.substr(intentURL.indexOf('=') + 1);
```

Comment:
Getting an error on this line when working step 1 of [V2 on this task](https://www-demo4.allizom.org/en-US/contribute/task/follow-mozilla/?v=2). Maybe a conditional needed around `$this.data('action')`?